util/uuid.c

```diff
@@ -1,34 +1,33 @@
-#include <uuid.h>
 #include "util/uuid.h"
 
-#if HAS_LIBUUID
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uuid_generate_random(uuid);
-	uuid_unparse(uuid, out);
-	return true;
-}
-#else
-#include <assert.h>
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
 #include <string.h>
-#include <stdlib.h>
 
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uint32_t status;
-	uuid_create(&uuid, &status);
-	if (status != uuid_s_ok) {
+#define UUID_FORMAT "%04" PRIx16 "%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 "%04" PRIx16 "%04" PRIx16
+bool generate_uuid(char out[static UUID_STRLEN])
+{
+	static FILE *urandom = NULL;
+	uint16_t data[8];
+
+	if (!urandom) {
+		if (!(urandom = fopen("/dev/urandom", "r"))) {
+			return false;
```

Comment:
It would be nice to call `wlr_log_errno` here to make it clear something wrong happened.